vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -499,11 +499,12 @@ func (feeder *clusterStateFeeder) LoadRealTimeMetrics(ctx context.Context) {
 	sampleCount := 0
 	droppedSampleCount := 0
 	for _, containerMetrics := range containersMetrics {
-		podInitContainers := feeder.clusterState.Pods()[containerMetrics.ID.PodID].InitContainers
-		if slices.Contains(podInitContainers, containerMetrics.ID.ContainerName) {
-			klog.V(3).InfoS("Skipping metric samples for init container", "pod", klog.KRef(containerMetrics.ID.PodID.Namespace, containerMetrics.ID.PodID.PodName), "container", containerMetrics.ID.ContainerName)
-			droppedSampleCount += len(containerMetrics.Usage)
-			continue
+		if pod, exists := feeder.clusterState.Pods()[containerMetrics.ID.PodID]; exists && pod != nil {
```

Comment:
Can we add a similar comment as below? Something like:

`// Not all pod states are tracked in memory saver mode`

Just helps give context behind the defensive check here.

Reply:
I've applied the suggestion. PTAL when you can :)